client/wsclient.go

```diff
@@ -85,7 +87,41 @@ func (c *wsClient) Stop(ctx context.Context) error {
 	conn := c.conn
 	c.connMutex.RUnlock()
 
+	ticker := time.NewTicker(50 * time.Millisecond)
+
+	// Wait for all remaining messages to be sent. Continuing to attempt
+	// to send messages after calling Stop will eventually result in lost
+	// messages.
+	select {
+	case <-ticker.C:
+		if !c.sender.NextMessage().IsPending() {
+			break
+		}
+	case <-time.After(3 * time.Second):
```

Comment:
Is this ever going to fire? The first case is always going to happen first, right?

Reply:
Realistically the first case should always happen first, yes. The most likely situation where we would hit this case is if the client implementation was continuously scheduling messages to be sent to the server and the sender thread wasn't processing them quickly enough for us to hit this case.

Reply:
I don't understand. How can this second case ever get selected? There are 2 timers, one fires in 50ms, the other in 3 seconds. What sort of timer discrepancy do you expect that selects the second case before the first case's channel is ready for reading?

I am probably missing something but it seems what effectively lines are 90-104 do is sleep for 50ms.

Reply:
>I inexplicably convinced myself there was a loop around this

I was suspecting this :-) I should have mentioned this in my comment, my fault.